fecgov/fec-cms
# Upgrade ESLint, resolve errors

## Summary

- Resolves #3125 
Upgraded ESLint to address the **critical** vulnerability, but the new version had several new warnings so I addressed those, too.

## Impacted areas of the application
Nearly every instance of JavaScript across the site, though it's only an eslint upgrade, which is only used when verifying our code format is to-spec. This upgrade removed a lot of global decorations we'd used across the site, which _could_ impact everything—unlikely, but possible.

## Screenshots
None

## Related PRs
None

## How to test
- Pull the branch
- `npm i`
- `npm run test-single` check for warnings (should test fine)
- `npm run build` (just in case)
- Run the server and check JavaScript modules across the site. data, cms, legal, glossary, calendar, feedback, datatables, etc. Basically everything even slightly interactive. _But_ checking one type of interaction should be fine for that category (e.g., if one datatable works, all the others that look like it probably will, too)
____
